Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM node:13 AS build_fe
+
+WORKDIR /usr/src/app
+COPY apsi-enrollment-fe/package.json apsi-enrollment-fe/
+WORKDIR /usr/src/app/apsi-enrollment-fe
+RUN npm install
+RUN npm install -g @angular/cli
+
+COPY apsi-enrollment-fe .
+RUN ng build --prod
+
+
+FROM gradle:6.3-jdk11 AS build_be
+
+WORKDIR /usr/src/app
+COPY apsi-enrollment-be .
+
+COPY --from=build_fe /usr/src/app/apsi-enrollment-fe/dist/apsi-enrollment-fe /usr/src/app/src/main/resources/static
+
+RUN gradle wrapper
+RUN ./gradlew build
+
+FROM openjdk:11-jre-slim
+COPY --from=build_be /usr/src/app/build /usr/src/app
+COPY --from=build_be /usr/src/app/build/libs/*.jar /usr/src/app/libs/apsi-enrollment.jar
+
+EXPOSE 8080
+CMD [ "java", "-jar", "/usr/src/app/libs/apsi-enrollment.jar" ]
```

Comment:
Ponawiam moje pytanie czy nie lepiej zastąpić ten dockerfile prostym composem. Teraz jeżeli będziemy chcieli wprowadzić jakąś zmianę do budowania dockerów trzeba to zrobić w przynajmniej 2 miejscach.

Nie wiem co prawda jak to się będzie miało do deploya.

Reply:
Można, dlatego są też 2 Dockerfile oddzielnie dla FE i BE. Deploy będzie konfigurowany w ramach innego zadania.